onnxruntime/core/providers/xnnpack/nn/max_pool.cc

```diff
@@ -132,15 +134,33 @@ MaxPool::MaxPool(const OpKernelInfo& info)
                   inferred_output_shape[3] == output_dims_[3],
               "Shape mismatch between inferred value and calculated value.");
 
-  xnn_status status;
-  struct xnn_operator* p;
-  status = xnn_create_max_pooling2d_nhwc_f32(input_padding_top, input_padding_right,
-                                             input_padding_bottom, input_padding_left,
-                                             pooling_height, pooling_width,
-                                             stride_height, stride_width,
-                                             dilation_height, dilation_width,
-                                             C, C, C,  // channels, input_pixel_stride, output_pixel_stride
-                                             output_min, output_max, flags, &p);
+  xnn_status status = xnn_status_invalid_state;
+  struct xnn_operator* p = nullptr;
+  if (X_arg.TypeAsProto()->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+    maxpool_type_ = OpComputeType::op_compute_type_fp32;
+    float output_min = clip_min_max_ ? clip_min_max_->first : -INFINITY;
+    float output_max = clip_min_max_ ? clip_min_max_->second : INFINITY;
+    status = xnn_create_max_pooling2d_nhwc_f32(input_padding_top, input_padding_right,
+                                               input_padding_bottom, input_padding_left,
+                                               pooling_height, pooling_width,
+                                               stride_height, stride_width,
+                                               dilation_height, dilation_width,
+                                               C, C, C,  // channels, input_pixel_stride, output_pixel_stride
+                                               output_min, output_max, flags, &p);
+  } else if (X_arg.TypeAsProto()->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType_UINT8) {
+    maxpool_type_ = OpComputeType::op_compute_type_qu8;
+    uint8_t output_min = clip_min_max_ ? static_cast<uint8_t>(clip_min_max_->first) : 0;
+    uint8_t output_max = clip_min_max_ ? static_cast<uint8_t>(clip_min_max_->second) : 255;
+    status = xnn_create_max_pooling2d_nhwc_u8(input_padding_top, input_padding_right,
+                                              input_padding_bottom, input_padding_left,
+                                              pooling_height, pooling_width,
+                                              stride_height, stride_width,
+                                              dilation_height, dilation_width,
+                                              C, C, C,  // channels, input_pixel_stride, output_pixel_stride
+                                              output_min, output_max, flags, &p);
```

Comment:
As long as It's DQ and Q have the same scale and zero_point, we can drop it's DQ and Q nodes.
Because, there is no computation in this OP, and It won't introduce any errors.